Sources/FDB/AnyFDBTransaction.swift

```diff
@@ -291,11 +292,19 @@ public protocol AnyFDBTransaction {
     /// - returns: EventLoopFuture with future Int64 value
     func getReadVersion() -> EventLoopFuture<Int64>
 
-    /// Returns versionstamp which was used by any versionstamp operations in this transaction
+    /// Returns a future for the versionstamp which was used by any versionstamp operations
+    /// in this transaction. Unlike the synchronous version of the same name, this method
+    /// does _not_ commit by default.
     ///
     /// - returns: EventLoopFuture with future FDB.Versionstamp value
     func getVersionstamp() -> EventLoopFuture<FDB.Versionstamp>
 
+    /// Returns a future for the versionstamp which was used by any versionstamp operations
+    /// in this transaction, and optionally commit the transaction right afterwards
+    ///
+    /// - returns: EventLoopFuture with future FDB.Versionstamp value
+    func getVersionstamp(commit shouldCommit: Bool) -> EventLoopFuture<FDB.Versionstamp>
```

Comment:
Should this return the transaction as well? I noticed other methods do this…
```suggestion
    func getVersionstamp(commit shouldCommit: Bool) -> EventLoopFuture<(FDB.Versionstamp, AnyFDBTransaction)>
```

What about `func getVersionstamp() -> EventLoopFuture<FDB.Versionstamp>`?

Reply:
Oof. Tough question. I gotta think about it.

Reply:
On the plus side, I'm likely the only user of Versionstamps right now, so probably no one will notice if the return signature changes a patch release in 🙃

Reply:
Yes, as for this feature, semver stability doesn't really bother me right now :) Yes, let's return a tuple of a versionstamp and a transaction, for the sake of consistency.

Reply:
One more data point: `get` has both signatures: 
<img width="468" alt="image" src="https://user-images.githubusercontent.com/225505/91601845-31623880-e91f-11ea-9da1-2fef95477884.png">

Reply:
Actually, it can be solved by adding the following to the extension on AnyFDBTransaction:
```swift
    /// Returns a future for the versionstamp which was used by any versionstamp operations
    /// in this transaction. Unlike the synchronous version of the same name, this method
    /// does _not_ commit by default.
    ///
    /// - returns: EventLoopFuture with future FDB.Versionstamp value
    func getVersionstamp() -> EventLoopFuture<(FDB.Versionstamp, AnyFDBTransaction)> {
        return self
            .getVersionstamp()
            .map { ($0, self) }
    }

    /// Returns a future for the versionstamp which was used by any versionstamp operations
    /// in this transaction, and optionally commit the transaction right afterwards
    ///
    /// - returns: EventLoopFuture with future FDB.Versionstamp value
    func getVersionstamp(commit shouldCommit: Bool) -> EventLoopFuture<(FDB.Versionstamp, AnyFDBTransaction)> {
        return self
            .getVersionstamp(commit: shouldCommit)
            .map { ($0, self) }
    }
```

Reply:
Wellllll, I'm afraid, it was done out of semver desperation or smth, I've had much more pain from these two definitions than convenience.

From the other side... Some use transactions directly, some with wrapper... Dammit. I gotta sleep with this dilemma. Don't merge pls, I haven't decided yet.

Reply:
I don't have a merge button, so you are in luck 😉

With the snippet I posted above, the user has both variations at their disposal, and it seems like the type system is smart enough to differentiate between then, seeing as I didn't need to touch the tests...

(though they are casting to begin with since it was necessary to get the rest of the type system happy anyways:
```swift
                    .flatMap { _ -> EventLoopFuture<FDB.Versionstamp> in
                        return transaction.getVersionstamp(commit: true)
                    }
```

Reply:
That's weird, I approved your PR, so you should be able to merge it. Funny.
Ye, I like two versions with different arguments, let's do that.

Reply:
(I believe it's because I don't have write access, which I absolutely don't need 😉)

Changes have been pushed!